include/oneapi/dpl/pstl/hetero/dpcpp/unseq_backend_sycl.h

```diff
@@ -114,56 +114,32 @@ struct walk_adjacent_difference
 // transform_reduce
 //------------------------------------------------------------------------
 
-// calculate shift where we should start processing on current item
-template <typename _NDItemId, typename _GlobalIdx, typename _SizeNIter, typename _SizeN>
-_SizeN
-calc_shift(const _NDItemId __item_id, const _GlobalIdx __global_idx, _SizeNIter& __n_iter, const _SizeN __n)
-{
-    auto __global_range_size = __item_id.get_global_range().size();
-
-    auto __start = __n_iter * __global_idx;
-    auto __global_shift = __global_idx + __n_iter * __global_range_size;
-    if (__n_iter > 0 && __global_shift > __n)
-    {
-        __start += __n % __global_range_size - __global_idx;
-    }
-    else if (__global_shift < __n)
-    {
-        __n_iter++;
-    }
-    return __start;
-}
-
 template <typename _ExecutionPolicy, typename _Operation1, typename _Operation2>
 struct transform_init
 {
     _Operation1 __binary_op;
     _Operation2 __unary_op;
 
-    template <typename _NDItemId, typename _GlobalIdx, typename _Size, typename _AccLocal, typename... _Acc>
+    template <typename _NDItemId, typename _Size, typename _ItersPerWG, typename _AccLocal, typename... _Acc>
     void
-    operator()(const _NDItemId __item_id, const _GlobalIdx __global_idx, _Size __n, _AccLocal& __local_mem,
+    operator()(const _NDItemId __item, _Size __n, _ItersPerWG __iters_per_witem, _AccLocal& __local_mem,
```

Comment:
Variable `__iters_per_witem` with `_ItersPerWG` type looks confusing, let's change the naming: `_ItersPerWG` -> `_ItersPerWItem`.

Reply:
Sure

Reply:
Sorry, first of all initially the naming was "per work group" rather then "per work item". Please make sure that the naming reflects what it should.

Second, I would not save 3 letters `_ItersPerWorkItem` looks much better. I don't suggest to change `_NdItemId` because this term comes from DPC++ directly.